### PR TITLE
fix(vrl): replace referrer by referer on nginx parser

### LINKED ISSUE #7726: Nginx VRL parsing function does not correctly parse some Nginx error logs
### Vector Version

```
vector 0.13.1 (v0.13.1 x86_64-apple-darwin 2021-04-29)
```

### Expected Behavior

The `parse_nginx_log(.message, "error")` function should correctly parse any nginx error log.

### Actual Behavior

In the following example log the `referrer` key is not recognized at all and `host` is equal to `65.21.190.83:31256", referrer: "http://65.21.190.83:31256/`.

### Example Data

```
2021/06/03 09:30:50 [error] 32#32: *6 open() "/usr/share/nginx/html/favicon.ico" failed (2: No such file or directory), client: 10.244.0.0, server: localhost, request: "GET /favicon.ico HTTP/1.1", host: "65.21.190.83:31256", referrer: "http://65.21.190.83:31256/"
```

### References

The Nginx implementation is here: https://github.com/nginx/nginx/blob/master/src/http/ngx_http_request.c#L3811 - you can see that it's not just `referrer` that is not accounted for in https://github.com/timberio/vector/blob/master/lib/vrl/stdlib/src/log_util.rs#L98, it's also `subrequest` and `upstream`.

### PR DESCRIPTION
**Pull request by jdrouet**
fixes #7726 comment

Signed-off-by: Jérémie Drouet <jeremie.drouet@datadoghq.com>

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
👍 thanks! As long as we are consistent, I don't have super strong opinions about the naming. I know the HTTP header calls it the misspelled `referer`

**Comment by spencergilbert** [comment]
> 👍 thanks! As long as we are consistent, I don't have super strong opinions about the naming. I know the header calls it the misspelled `referer`

I feel like... sticking to the header name is least surprising? Just add to the docs the reasoning why?

**Comment by jszwedko** [comment]
> > 👍 thanks! As long as we are consistent, I don't have super strong opinions about the naming. I know the header calls it the misspelled `referer`
> 
> I feel like... sticking to the header name is least surprising? Just add to the docs the reasoning why?

Yeah, I have a slight leaning that way too -- to use `referer` -- given this is widely used for [historical reasons](https://en.wikipedia.org/wiki/HTTP_referer#Etymology).

**Comment by jdrouet** [comment]
The problem with using `referer` instead of `referrer` is that nginx itself, in the error lines, uses `referrer`. So when we parse the `key: value`s we'll end up having `referer` from the `combined` logs and `referrer` from the error logs. Or we have to add a condition when parsing 🤔

**Comment by spencergilbert** [comment]
> The problem with using `referer` instead of `referrer` is that nginx itself, [in the error lines](https://github.com/timberio/vector/blob/master/lib/vrl/stdlib/src/parse_nginx_log.rs#L249), uses `referrer`. So when we parse the `key: value`s we'll end up having `referer` from the `combined` logs and `referrer` from the error logs. Or we have to add a condition when parsing 🤔

Open a bug against NGINX! 😃

**Comment by jdrouet** [comment]
I'll try to open a PR changing [this code](https://github.com/nginx/nginx/blob/07c63a42640e59bf5e3399cfdafd498b61671780/src/http/ngx_http_request.c#L3892)

**Comment by jdrouet** [comment]
It appears that I finally managed to send the patch with mercurial, but in the meantime, did a fix to automatically rename `referrer` to `referer`.

File: lib/vrl/stdlib/src/log_util.rs [hunk 1]
```diff
@@ -88,7 +88,7 @@ lazy_static! {
         )"\s+                                   # Match any non space character
         (?P<status>\d+)\s+                      # Match numbers
         (?P<size>\d+)\s+                        # Match numbers
-        "(-|(?P<referer>[^"]+))"\s+             # Match `-` or any non double-quote character
+        "(-|(?P<referrer>[^"]+))"\s+            # Match `-` or any non double-quote character
```
Comment (spencergilbert) [inline lib/vrl/stdlib/src/log_util.rs:91]: Shouldn't this one remain as `referer`?

Comment (jdrouet) [reply lib/vrl/stdlib/src/log_util.rs:91]: yeah, why did I do that? 🤔

Comment (spencergilbert) [reply lib/vrl/stdlib/src/log_util.rs]: 🤷

**Review by spencergilbert — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
🎉

**Pull request merged**